agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -575,8 +585,20 @@ bool monitor_thread::update_sta_stats()
             std::chrono::duration_cast<std::chrono::milliseconds>(now - sta_stats.last_update_time);
         sta_stats.delta_ms         = float(time_span.count());
         sta_stats.last_update_time = now;
+
+        if (std::chrono::steady_clock::now() > awake_timeout()) {
+            // If we haven't finish to iterate on all stations, set the select timeout to minimal
+            // value (1 msec) temporarily so the select will not be stuck on full select timeout
+            // and the thread will be able to finish this operation quickly.
+            set_select_timeout(1, false);
+            return true;
+        }
     }
 
+    // Reset select timeout value to init value
+    set_select_timeout(0);
```

Comment:
a bit confusing , if you want reset the TO just add `reset_select_timeout`.

Reply:
I changed it so it would be done automatically by the socket thread.